File: examples/quickstart-microsoft-browsers/src/main/java/io/fluentlenium/examples/pages/DuckDuckMainPage.java
package io.fluentlenium.examples.pages;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.concurrent.TimeUnit;

import io.fluentlenium.core.FluentPage;
import io.fluentlenium.core.annotation.PageUrl;
import io.fluentlenium.core.domain.FluentWebElement;
import org.openqa.selenium.support.FindBy;

@PageUrl("https://duckduckgo.com")
public class DuckDuckMainPage extends FluentPage {
    private static final String SEARCH_FORM_HOMEPAGE = "#search_form_homepage";

    @FindBy(css = "#search_form_input_homepage")
    private FluentWebElement searchInput;

    @FindBy(css = "#search_button_homepage")
    private FluentWebElement searchButton;

    public DuckDuckMainPage typeSearchPhraseIn(String searchPhrase) {
        await().until(searchInput).displayed();
        searchInput.write(searchPhrase);
        return this;
    }

    public DuckDuckMainPage submitSearchForm() {
        searchButton.submit();
        awaitUntilSearchFormDisappear();
        return this;
    }

    public void assertIsPhrasePresentInTheResults(String searchPhrase) {
        assertThat(window().title()).contains(searchPhrase);
    }

    private DuckDuckMainPage awaitUntilSearchFormDisappear() {
        await().atMost(5, TimeUnit.SECONDS).until(el(SEARCH_FORM_HOMEPAGE)).not().present();
        return this;
    }
}

File: examples/quickstart-firefox/src/main/java/org/fluentlenium/examples/pages/basic/DuckDuckMainPage.kt
package org.fluentlenium.examples.pages.basic

import org.assertj.core.api.Assertions.assertThat
import org.fluentlenium.core.FluentPage
import org.fluentlenium.core.annotation.PageUrl
import org.fluentlenium.core.domain.FluentWebElement
import org.openqa.selenium.support.FindBy
import java.util.concurrent.TimeUnit

@PageUrl("https://duckduckgo.com")
class DuckDuckMainPage : FluentPage() {

    @FindBy(css = "#search_form_input_homepage")
    private lateinit var searchInput: FluentWebElement

    @FindBy(css = "#search_button_homepage")
    private lateinit var searchButton: FluentWebElement

    fun typeSearchPhraseIn(searchPhrase: String?): DuckDuckMainPage {
        searchInput.write(searchPhrase)
        return this
    }

    fun submitSearchForm(): DuckDuckMainPage {
        searchButton.submit()
        awaitUntilSearchFormDisappear()
        return this
    }

    fun assertIsPhrasePresentInTheResults(searchPhrase: String?) {
        assertThat(window().title()).contains(searchPhrase)
    }

    private fun awaitUntilSearchFormDisappear(): DuckDuckMainPage {
        await().atMost(5, TimeUnit.SECONDS).until(el(SEARCH_FORM_HOMEPAGE)).not().present()
        return this
    }

    companion object {
        private const val SEARCH_FORM_HOMEPAGE = "#search_form_homepage"
    }
}